src/OrleansRuntime/Silo/Silo.cs

```diff
@@ -280,6 +285,16 @@ internal Silo(SiloInitializationParameters initializationParams)
             services.AddSingleton<DefaultPlacementStrategy>();
             services.AddSingleton<ClientObserversPlacementDirector>();
 
+            // Versions
+            services.AddSingleton<VersionSelectorManager>();
```

Comment:
I suppose it takes the first and it works... I will merge the ctor, not needed to have two here anyway